api-guidelines/v4.0/API-DESIGN-GUIDELINES/RESOURCES.md

```diff
@@ -57,43 +67,83 @@ School Year, and a reference to a School. Resources _must_ be accessible by
+uniquely describe a single EducationOrganizationAssociation.
 
-The Ed-Fi UDM describes cases in which fields on a resource are merged with
-other fields on a resource, producing a situation in which two fields must have
-the same value. This most commonly occurs in references to other API resources,
-as these references use the (often composite) primary key fields. In such cases,
-key fields from the reference may be defined — via the UDM — to be unified with
-existing fields on the resource.
-
-This unification has proven extremely valuable to ensuring basic data quality in
-field operations. All field and key unification scenarios defined in the
-UDM _must_ be implemented in an Ed-Fi REST API.
+```json
+{
+  "id": "bb82e4c1433547f3bedf97c133e8148a",
+  "educationOrganizationReference": {
+    "educationOrganizationId": 255901
+  },
+  "studentReference": {
+    "studentUniqueId": "604824"
+  },
+  ...
+}
+```
 
 ## Resource Extensions
 
 The Ed-Fi UDM can be extended to augment the structure of existing entities or
-create entirely new entities.[\[6\]](#f6)In the context of an Ed-Fi REST API,
-these are called resource extensions. Consumers of an Ed-Fi REST API interact
-with resource extensions just as they interact with other Resources. For
-example, if the Student resource has been extended in the data model supported
-by the API platform host, an API consumer requesting a student will receive the
-extended resource.
-
-An Ed-Fi REST API resource _must_ follow a specific pattern for extensibility in
-its structure in order to distinguish extension attributes from native
-attributes. Extensions _must_ be denoted by use of the reserved term `_ext` and
-namespaced. For API users, this both clearly distinguishes these attributes from
-native attributes that originate from the UDM, and provides (via the namespace)
+create entirely new entities (cf [Ed-Fi Data Standard Extension
+Framework](https://edfi.atlassian.net/wiki/x/LYBgAw)). In the context of an
+Ed-Fi REST API, these are called resource extensions. Consumers of an Ed-Fi REST
+API interact with resource extensions just as they interact with other
+Resources. For example, if the Student resource has been extended in the data
+model supported by the API platform host, an API consumer requesting a student
+will receive the extended resource.
+
+An Ed-Fi API resource _must_ follow a specific pattern for extensibility in its
+structure in order to distinguish extension attributes from native attributes.
+Extensions _must_ be denoted by use of the reserved term `_ext` and namespaced.
+For API users, this both clearly distinguishes these attributes from native
+attributes that originate from the UDM, and provides (via the namespace)
 information as to the origin or governance of the extension. The example below
-shows the JSON for a Staff resource that has been extended using a "_GrandBend_"
+shows the JSON for a Staff resource that has been extended using a `_grandbend_`
```

Comment:
I'm not sure I quite like the word "namespace" here mostly because it has nothing to do with namespaces in the descriptor sense (seemingly) and just overloads the term.

Reply:
It is the more generic sense of namespace. Probably not something I can lightly change, because this is terminology that is widespread in documentation on writing extensions and probably embedded in code as well. Still, I'm not resolving this comment yet - I'll sit with the idea for a bit to see if anything comes to me.